addon/initializers/asset-map.js

```diff
@@ -0,0 +1,59 @@
+import fetch from 'fetch';
+import { isNone } from '@ember/utils';
+import AssetMap from 'ember-theme-changerr/services/asset-map';
+import { warn } from '@ember/debug';
+
+export function initialize(app) {
+  // if we're in an engine return early as engines
+  // don't need to defer their readiness.
+
+  if (!app.deferReadiness) {
+    return;
+  }
+
+  app.deferReadiness();
+  const config = app.resolveRegistration('config:environment');
+
+  const useAssetMap = (isNone(config.theme) && isNone(config.theme.useAssetMap)) ?
+    config.environment === 'production' :
+    config.theme.useAssetMap;
+
+  if (!useAssetMap) {
+    app.register('service:asset-map', AssetMap);
+    app.advanceReadiness();
+    return;
+  }
+
+  fetch('/assets/assetMap.json', {
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  }).then((response) => {
+    if (response.ok) {
+      return response.json();
+    }
+    throw new Error('Network response was not ok.');
+  }).then((map) => {
+    const prepend = config.theme.assetPrepend || map.prepend;
+    const EnabledAssetMap = AssetMap.extend({
+      assetMapHash: map.assets,
+      prepend,
+      enabled: true
+    });
```

Comment:
Re-opening the original class was causing test failures as it changed the class across all tests. So I changed it to use `extend()`

I don't imagine this would impact regular applications as `AssetMap` would only get initialized once